src/main/java/bio/terra/app/controller/GlobalExceptionHandler.java

```diff
@@ -66,6 +67,7 @@ public ErrorModel conflictHandler(ErrorReportException ex) {
 
   // -- exceptions from validations - we don't control the exception raised --
   @ExceptionHandler({
+    TypeMismatchException.class,
```

Comment:
What about this change required this type to be added here? I would expect the UUID to be validated before it arrives at the controller code, as it is in other controllers in TDR.

For example, if I try using an invalid UUID against TDR running locally, I get:
```
$ curl http://localhost:8080/api/repository/v1/datasets/blah
{"message":"400 BAD_REQUEST - see error details","errorDetail":["Invalid UUID string: blah"]}
```

Reply:
I'm not sure why it is needed but the tests for invalid UUID fail without it.

Reply:
> I'm not sure why it is needed but the tests for invalid UUID fail without it.

This also rings a bell for me - I remember trying to boost the test coverage to ensure the methods handled the non-UUID case.